src/export/csv.rs

```diff
@@ -72,6 +73,14 @@ pub fn export_csv(exporter: &mut Exporter) -> Result<()> {
         let native = ctx.map(|&(_, ref n)| n).flatten();
 
         if let Some(curr) = foreign.curr {
+            match end_at {
+                Some(seconds) if curr.period.begin() > seconds => {
+                    println!("Ending at {:?} seconds of video.", seconds);
+                    break;
+                }
+                _ => {}
+            }
+
```

Comment:
Maybe an `if let` statement might allow you to drop the `_ => {}` at least?

```rust
fn main() {
    let v1 = Some("world");
    if let Some(v2) = v1 {
        if v2 == "world" {
            println!("Hello, world!");
        }
    }
}
```

Another option (probably a better one) is to move the duplicated code into a method on `exporter`, so you can just:

```rust
if exporter.finishedBefore(seconds) {
    break;
}
```

Or even more ambitiously:

```rust
if !exporter.shouldExportAt(seconds) {
    next;
}
```

This version would also work for a `--start-at` option, which is inevitably going to be added once `--end-at` exists. :-)

Also, as a general rule for `substudy`, it's often worth replacing `println!` (when useful) with `debug!`. This will hide the extra output unless the user sets `RUST_LOG=debug` in the environment, but I generally prefer commands that don't talk much as long as they're doing what you asked them to do. ;-) But I do like generous use of `debug!` and `trace!`; it's proven extraordinarily useful in production. The `log` + `env_logger` crates are pretty awesome for something so simple.